### PR TITLE
docs: add basic examples and clarify spec

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

File: doc/examples.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+# Example Configs
+
+Each of these examples is written in version 1 of the config. Ensure that any configuration matches the version that Ignition expects.
```
Comment (dghubble) [inline doc/examples.md]: written in version 2

Comment (crawford) [inline doc/examples.md]: shit

File: doc/configuration.md [hunk 2]
```diff
@@ -3,14 +3,14 @@
 The Ignition configuration is a JSON document conforming to the following specification, with **_italicized_** entries being optional:
 
 * **ignition** (object): metadata about the configuration itself.
-  * **version** (string): the semantic version number of the spec. Must be at least `2.0.0`.
```
Comment (dghubble) [inline doc/configuration.md]: Can this page have a link to the last Ignition v1 docs? Or do it right before v2.

Comment (crawford) [inline doc/configuration.md]: I'm going to merge the v2 branch into master, so it should be atomic.

File: doc/examples.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+```json
+{
+  "ignition": { "version": "2.0.0" },
+  "storage": {
+    "files": [{
+      "filesystem": "root",
+      "path": "/foo/bar",
+      "contents": {
+        "source": "http://example.com/asset",
+        "verification": { "hash": "sha512-0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef" }
```
Comment (dghubble) [inline doc/examples.md]: Is whitespace trimming being done before hashing. Might be useful to show the command line equivalent for getting the right hash. Saves people a cycle of getting it wrong and updating.

Comment (crawford) [inline doc/examples.md]: There is zero modification done to the file. Good idea though.

**Comment by dghubble** [comment]
Seems good to me

**Pull request merged**